haystack/nodes/connector/crawler.py

```diff
@@ -164,38 +164,40 @@ def __init__(
             logger.info("'chrome-driver' will be automatically installed.")
             self.driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()), options=options)
         self.urls = urls
-        self.output_dir = output_dir
         self.crawler_depth = crawler_depth
         self.filter_urls = filter_urls
         self.overwrite_existing_files = overwrite_existing_files
         self.id_hash_keys = id_hash_keys
         self.extract_hidden_text = extract_hidden_text
         self.loading_wait_time = loading_wait_time
         self.crawler_naming_function = crawler_naming_function
+        self.output_dir = output_dir
+        self.file_path_meta_field_name = file_path_meta_field_name
 
     def __del__(self):
         self.driver.quit()
 
     def crawl(
         self,
-        output_dir: Union[str, Path, None] = None,
         urls: Optional[List[str]] = None,
         crawler_depth: Optional[int] = None,
         filter_urls: Optional[List] = None,
-        overwrite_existing_files: Optional[bool] = None,
         id_hash_keys: Optional[List[str]] = None,
         extract_hidden_text: Optional[bool] = None,
         loading_wait_time: Optional[int] = None,
+        output_dir: Union[str, Path, None] = None,
+        overwrite_existing_files: Optional[bool] = None,
+        file_path_meta_field_name: Optional[str] = None,
         crawler_naming_function: Optional[Callable[[str, str], str]] = None,
-    ) -> List[Path]:
+    ) -> List[Document]:
         """
         Craw URL(s), extract the text from the HTML, create a Haystack Document object out of it and save it (one JSON
         file per URL, including text and basic meta data).
         You can optionally specify via `filter_urls` to only crawl URLs that match a certain pattern.
         All parameters are optional here and only meant to overwrite instance attributes at runtime.
         If no parameters are provided to this method, the instance attributes that were passed during __init__ will be used.
 
-        :param output_dir: Path for the directory to store files
+
```

Comment:
```suggestion
```
Sorry for being picky :), this text is translated to markdown and newlines matter when they ultimately hit readme.io for rendering